File: exercicios em C/q22.cpp
#include <stdio.h>

int main(void){
	int h, i, n;
	h = 0;
	printf("\n\nEste programa gerar o numero H, levando em conta que o usuario decide a quantidade valores.");
	printf("\n\nDigite a quantidade de valores para chegar no numero H: ");
	scanf("%d", &n);
	while (n > 0){
		h = h + 1 / n;
		n--;
	}
	
	//Copiei de um exemplo seu, nao fasso ideia se ta certo ou nao;-;
	
	printf("\n\nO valor de H e: %d", h);
	printf("\n\nObrigado por utilizar nosso programa!");
	return 0;
}

File: exercicios em Python/q22.py
import os
h = 0
print("\n\nEste programa gerar o numero H, levando em conta que o usuario decide a quantidade valores.")
n = int (input("\n\nDigite a quantidade de valores para chegar no numero H: "))
while (n > 0):
        h =int (h + 1 / n)
        n = n - 1
        
print("\n\nO valor de H e: ", h)
print("\n\nObrigado por utilizar nosso programa!")
os.system("pause")
